src/components/Contact/Contact.jsx

```diff
@@ -0,0 +1,114 @@
+import { useState, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import useTranslation from 'hooks/useTranslation';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { useContactMutation } from 'services/contact/contact';
+import Input from 'components/form/Input/Input';
+import Button from 'components/common/Button/Button';
+import close from 'assets/close.svg';
+import smiles from 'assets/smiles.svg';
+import sad from 'assets/sad.svg';
+
+const Contact = ({ isContactOpen, handleContact }) => {
+  const t = useTranslation();
+  const [contact, { isLoading, isSuccess, error }] = useContactMutation();
+  const schema = z.object({
+    email: z.string().email({ message: t('contact.errors.emailMsg') }),
+    body: z.string().min(1, { message: t('contact.errors.required') }),
+  });
+  const [feedback, setFeedback] = useState({ success: false, loading: false, error: false });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: zodResolver(schema) });
+
+  const onSubmit = data => {
+    contact(data);
+  };
+
+  const handleFeedback = () => {
+    if (isLoading) {
+      setFeedback({ ...feedback, loading: true });
```

Comment:
since you rely on the previous state I'll suggest you to
```suggestion
      setFeedback(previous => { ...previous, loading: true });
```

Reply:
I don't understand why is it better to do it this way. Can you tell me?

Reply:
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous